pkg/traceroute/udp4.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package traceroute
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"math/rand"
+	"net"
+	"time"
+
+	"golang.org/x/net/ipv4"
+)
+
+// SendTrace in a routine
+func (t *Trace) SendTracesUDP4() {
```

Comment:
so if Trace were an interface, you'd have func (t Trace) Send(), and the question of 4 or 6 or whatever would not have to appear in names everywhere. Or, you can have a net.Conn in your Trace struct and wrap the differences in there. But this TCP|UDP 4|6 variation doesn't quite feel right.

Reply:
ICMP tracing uses TCPProbe/Ping of the `god struct` tracer. I'm not happy with the implementation too, but I also having major issues finding a better/cleaner solution.